Tests/SwiftSyntaxTest/SyntaxConvenienceMethodsTests.swift

```diff
@@ -0,0 +1,32 @@
+import XCTest
+import SwiftSyntax
+
+public class SyntaxConvenienceMethodsTests: XCTestCase {
+
+  public func testFloatingValues() {
+    testFloatingValue(text: "5.3_8", expectedValue: 5.38)
+    testFloatingValue(text: "12e3", expectedValue: 12000.0)
+    testFloatingValue(text: "32E1", expectedValue: 320.0)
+    testFloatingValue(text: "0xdEFACE.C0FFEEp+1", expectedValue: 29226397.507810354)
+    testFloatingValue(text: "0xaffab1e.e1fP-2", expectedValue: 46131911.72064209)
```

Comment:
I could have used the following approach to avoid repetition of the `testFloatingValue()` call:

```
[
  ("5.3_8", 5.38),
  ("12e3", 12000.0),
  ("32E1", 320.0),
  ("0xdEFACE.C0FFEEp+1", 29226397.507810354),
  ("0xaffab1e.e1fP-2", 46131911.72064209)
].forEach {
  testFloatingValue(text: $0.0, expectedValue: $0.1)
}
```
Although this approach would have been [DRY](https://dzone.com/articles/software-design-principles-dry-and-kiss)er, I find that it makes failure diagnosis more difficult, so I did not use it.

Reply:
The testing approach seems good to me. One idea though (I’m not quite sure if it’s an improvement or not) would be to use the literal that you’re specifying in the String also in the Swift source code, i.e.\

```swift
testFloatingValue(text: "0xaffab1e.e1fP-2", expectedValue: 0xaffab1e.e1fP-2)
```

Mostly because my mental hex parsing skills aren't brilliant and I’m not quite sure if `0xaffab1e.e1fP-2` is really equal to `46131911.72064209`.

Reply:
Done. As an aside, [this](https://nedbatchelder.com/text/hexwords.html) is my go-to source for fun hex words.